splinepy/bezier.py

```diff
@@ -4,6 +4,61 @@
 class BezierBase(spline.Spline):
     """Bezier Base. Contain extra operations that's only
     available for bezier families.
+
+    Beziers are a special type of spline where the basis functions have
+    a global support. The basis functions are given by the so-called
+    *Berstein polynomials*
+
+    .. math::
+            B_{i;p}(u) = \\binom{p}{i} u^i (1-u)^{p-i}
+
+    The parametric domain of a Bezier spline is always a multi-dimensional
+    hypercube, that is :math:`\\Omega_{param}=[0,1]^{N_{param}}`.
+    We can distinguish between different types of splines depending on the
+    dimension of the parametric space.
+
+    1. A spline of degree :math:`p` with control points
+    :math:`P_i\\in\\mathbb{R}^{N_{phys}}` and a one-dimensional parameter space
+    (i.e., :math:`N_{param}=1`) corresponds to a line embedded into the
+    physical space:
+
+    .. math::
+            S^B(u) = \\sum_{i=0}^{l} B_{i;p}(u) P_i
+
+    2. A spline of degrees :math:`p,q` with control points
+    :math:`P_{i,j}\\in\\mathbb{R}^{N_{phys}}` and a two-dimensional parameter
+    space (i.e., :math:`N_{param}=2`) corresponds to a surface, embedded into
+    the physical space:
+
+    .. math::
+            S^B(u,v) = \\sum_{i=0}^{l} \\sum_{j=0}^{m} B_{i;p}(u) B_{j;q}(v)
+                P_{i,j}
+
+    Due to the tensor-product nature of the Bezier basis functions, this is
+    often rewritten as in terms of multi-variate basis functions
+
+    .. math::
+            \\tilde{B}_{i,j;p,q}(u,v) := B_{i;p}(u) B_{j;q}(v)
+
+    3. A spline of degrees :math:`p,q,r` with control points
+    :math:`P_{i,j,k}\\in\\mathbb{R}^{N_{phys}}` and a three-dimensional
+    parameter space (i.e., :math:`N_{param}=3`) corresponds to a volume,
+    embedded into the physical space:
+
+    .. math::
+            S^B(u,v,w) = \\sum_{i=0}^{l} \\sum_{j=0}^{m} \\sum_{k=0}^{n}
+                B_{i;p}(u) B_{j;q}(v) B_{k;r}(w) P_{i,j,k}
+
+    Here, we can introduce the multi-variate basis functions
+
+    .. math::
+            \\tilde{B}_{i,j,k;p,q,r}(u,v,w) := B_{i;p}(u) B_{j;q}(v)
+                B_{k;r}(w)
+
+    Finally, we like to emphasize that Bezier splines can also be seen as a
+    special type of B-Splines with open knot vectors (i.e., the first and last
+    entry are repeated :math:`p+1`-times) that do not feature additional
+    internal knots.
```

Comment:
What do you think of adding a simple example to show how it is to be used:
```suggestion
    internal knots.
    
    Usage:
    
    .. code-block:: python
       
       # Polynomial Bezier surface
       polynomial_bezier = splinepy.Bezier(
           degrees=[2,1],
           control_points=[
               [0.0, 0.0],
               [1.0, 0.0],
               [2.0, 1.0],
               [0.0, 2.0],
               [1.0, 1.0],
               [2.0, 2.0]
           ]
       )
       # Rational bezier surface (quarter arc)
       rational_bezier = splinepy.RationalBezier(
           degrees=[2],
           control_points=[
               [0.0, 0.0],
               [1.0, 0.0],
               [1.0, 1.0]
           ],
           weights=[1.0, 2.0**-0.5, 1.0]
       )
```

Reply:
Sure, I can do that, I'll copy examples from the examples folder.